src/Compiler/Checking/CheckFormatStrings.fs

```diff
@@ -299,32 +313,45 @@ let parseFormatStringInternal
 
     and parseSpecifier acc (i, fragLine, fragCol) fragments =
         let startFragCol = fragCol
-        let fragCol = fragCol+1
-        if fmt[i..(i+1)] = "%%" then
+        let nPercent =
+            fmt[i..]
+            |> Seq.takeWhile (fun c -> c = '%')
+            |> Seq.length
+        // TODO XXX assuming here that nPercent >= 1, should check?
```

Comment:
Is the comment still relevant?

Reply:
Decided to remove the comment without adding any checks. My reasoning is, right now the assumption is guaranteed to hold and it's IMO very unlikely to change with future PR, so not worth to complicate this code further